DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/twoarg.jl

```diff
@@ -1,32 +1,19 @@
-struct MooncakeTwoArgPullbackPrep{R,F,Y,DX,DY} <: PullbackPrep
-    rrule::R
-    df!::F
-    y_copy::Y
-    dx_righttype::DX
+struct MooncakeTwoArgPullbackPrep{Tcache,DY,F} <: PullbackPrep
+    cache::Tcache
     dy_righttype::DY
-    dy_righttype_after::DY
+    target_function::F
 end
 
 function DI.prepare_pullback(
     f!, y, backend::AutoMooncake, x, ty::NTuple, contexts::Vararg{Context,C}
 ) where {C}
-    config = get_config(backend)
-    rrule = build_rrule(
-        get_interpreter(),
-        Tuple{typeof(f!),typeof(y),typeof(x),typeof.(map(unwrap, contexts))...};
-        debug_mode=config.debug_mode,
-        silence_debug_messages=config.silence_debug_messages,
-    )
-    df! = zero_tangent(f!)
-    y_copy = copy(y)
-    dx_righttype = zero_tangent(x)
-    dy_righttype = zero_tangent(y)
+    target_function = function (f!, y, x, contexts...)
+        f!(y, x, contexts...)
+        return y
+    end
+    cache = prepare_pullback_cache(target_function, f!, y, x, map(unwrap, contexts)...)
```

Comment:
is there a reason why don't use the config here?

Reply:
No, that is a mistake on my part.

Reply:
Resolved by latest commit